src/json/jobs/JobParams.ts

```diff
@@ -0,0 +1,198 @@
+              items: {
+                type: "string",
+                enum: [
+                  "user.created",
+                  "coupon.created",
+                  "reward.created",
+                  "referral.started",
+                  "referral.converted",
+                  "user.reward.balance.changed",
+                  "email.referred.reward.earned",
+                  "email.referral.started",
+                  "email.referral.paid",
+                  "email.referral.rewardLimitReached",
+                  "referral.automoderation.complete",
+                  "referral.ended",
+                  "theme.publish.finished"
+                ]
+              }
+            }
+          },
+          additionalProperties: false,
+          required: ["enabledWebhookTypes"]
+        }
+      ],
+      default: true
+    },
+    programs: {
+      oneOf: [
+        {
+          type: "boolean",
+          title: "Program Evaluation Enabled",
+          description:
+            "If true then all applicable active programs will be triggered"
+        },
+        {
+          type: "object",
+          properties: {
+            enabledProgramIds: {
+              type: "array",
+              title: "Enabled Program Ids",
+              items: {
+                type: "string"
+              }
+            }
+          },
+          additionalProperties: false,
+          required: ["enabledProgramIds"]
+        }
+      ],
+      default: true
+    },
+    analytics: {
+      oneOf: [
+        { type: "boolean", title: "Analytics Tracking Enabled" },
+        {
+          type: "object",
+          properties: {
+            enabledAnalyticsEventCollections: {
+              type: "array",
+              title: "Analytics Event Collections",
+              items: {
+                type: "object",
+                properties: {
+                  collectionName: {
+                    type: "string",
+                    title: "Collection Name",
+                    enum: [
+                      "userCreated",
+                      "userReferralProgramLoaded",
+                      "userReferralProgramEngagement",
+                      "userReferralShareLinkClicked",
+                      "userReferralCreated",
+                      "userApprovedReferralCreated",
+                      "userReferralModerated",
+                      "userReferralConverted",
+                      "userApprovedReferralConverted",
+                      "userReferralEnded",
+                      "rewardCreated",
+                      "rewardRedeemed",
+                      "programEvaluated",
+                      "userActivity",
+                      "programGoal"
+                    ]
+                  },
+                  analyticTimestampSource: {
+                    type: "string", // TODO - probably going to delete this and only go with the historic analytic date if we can get it working
```

Comment:
@jericmason why is there still a `TODO` here?

Reply:
This should be removed as well.